### PR TITLE
CPUInfo fixes after #16098 

### PR DESCRIPTION
**Pull request by lrusak**
#16098 introduced a few issues with various platforms (some that jenkins doesn't even build)

This should fix some of them and some other issues that were found.

----
#### CPUInfoWin32: remove _M_ARM gaurd
 - not sure why this even existed in the past

----
#### CPUInfoLinux: return false if temperature units aren't valid
 - if a valid temperature isn't passed then we need to return false as the temperature won't be valid

----
#### CPUInfoLinux: fix ifdefs for non neon platforms

 this attempts to fix the build for all the following paths:
  - arm (without neon, ie RPi0/1)
  - arm (with neon, ie RPI2+)
  - aarch64 (doesn't need neon ?)
  - x86

----
#### CPUInfo: split posix functions into specific class to allow windows to build
 - this introduces an intermediate class CCPUInfoPosix which include posix specific commands to allow windows to build properly.

----
#### CCPUInfo: move servicebroker registration earlier in app startup
 - this fixes startup on RPi where it wants cpuinfo before the settingscomponent initialization

### CODE REVIEW

**Comment by lrusak** [comment]
jenkins build this please

**Comment by Rechi** [comment]
WIN-UWP-64 is still failing, because of #16098 and this PR, see build log.

File: xbmc/platform/win10/CPUInfoWin10.cpp [hunk 1]
```diff
@@ -117,3 +117,11 @@ int CCPUInfoWin10::GetUsedPercentage()
 
   return static_cast<int>(m_lastUsedPercentage);
 }
+
+bool CCPUInfoWin10::GetTemperature(CTemperature& temperature)
+{
+  temperature = CTemperature::CreateFromCelsius(-1);
+  temperature.SetValid(true);
```
Comment (Rechi) [inline xbmc/platform/win10/CPUInfoWin10.cpp]: why set the temperature to valid?

File: xbmc/platform/win10/CPUInfoWin10.cpp [hunk 2]
```diff
@@ -117,3 +117,11 @@ int CCPUInfoWin10::GetUsedPercentage()
...
+
+  return true;
```
Comment (Rechi) [inline xbmc/platform/win10/CPUInfoWin10.cpp]: why return true?

File: xbmc/platform/win10/CPUInfoWin10.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/platform/win10/CPUInfoWin10.cpp]: to allow it to pass the test

File: xbmc/platform/win10/CPUInfoWin10.cpp (hunk 2 above)
Comment (lrusak) [reply xbmc/platform/win10/CPUInfoWin10.cpp]: to allow it to pass the test

**Comment by lrusak** [comment]
jenkins build this please

File: xbmc/platform/win10/CPUInfoWin10.cpp (hunk 1 above)
Comment (Rechi) [reply xbmc/platform/win10/CPUInfoWin10.cpp]: Adding incorrect application code, to fix a wrong test case doesn't sound sane for me.

Comment (lrusak) [reply xbmc/platform/win10/CPUInfoWin10.cpp]: if you want to do the windows implementation be my guest

Comment (Rechi) [reply xbmc/platform/win10/CPUInfoWin10.cpp]:
```
bool CCPUInfoWin10::GetTemperature(CTemperature& temperature)
{
  temperature.SetValid(false);
  return false;
}
```

is the correct implementation.

**Comment by Rechi** [comment]
`TestCPUInfo.GetTemperature` is still failing on windows.
It would be failing on UWP too, but the test suite can't be run for that platform.

**Comment by lrusak** [comment]
jenkins build this please

**Comment by lrusak** [comment]
> `TestCPUInfo.GetTemperature` is still failing on windows.
> It would be failing on UWP too, but the test suite can't be run for that platform.

This should be fixed for windows. It should only fail on UWP now.

**Comment by Rechi** [comment]
It is still failing, I had already tested it with this PR. Here is the output.

```
kodi-test.exe --gtest_filter=TestCPUInfo.GetTemperature
Note: Google Test filter = TestCPUInfo.GetTemperature
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from TestCPUInfo
[ RUN      ] TestCPUInfo.GetTemperature
xbmc\utils\test\TestCPUInfo.cpp(51): error: Value of: CServiceBroker::GetCPUInfo()->GetTemperature(t)
  Actual: false
Expected: true
xbmc\utils\test\TestCPUInfo.cpp(52): error: Value of: t.IsValid()
  Actual: false
Expected: true
[  FAILED  ] TestCPUInfo.GetTemperature (275 ms)
[----------] 1 test from TestCPUInfo (275 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (711 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] TestCPUInfo.GetTemperature

 1 FAILED TEST
Clean shutdown of TestGlobalPattern1
```

**Comment by Rechi** [comment]
The `CCPUInfoWin32::GetTemperature` implementation was actually correct, only the test case is wrong.
The dummy implementation is also okay for me, because that's the same behaviour as before #16098.

**Comment by lrusak** [comment]
> The `CCPUInfoWin32::GetTemperature` implementation was actually correct, only the test case is wrong.
> The dummy implementation is also okay for me, because that's the same behaviour as before #16098.

That's fine with me. I've touched windows more than I would like to. If you are fine with how it is it would be nice to get this merged.

**Comment by lrusak** [comment]
jenkins build this please

**Comment by DaveTBlake** [comment]
Win32 test suite still not happy, see console output 
```
Running testsuite...
22:34:41 /xbmc/project/BuildDependencies/msys64/usr/bin/sed: can't read d:\jenkins\workspace\WIN-32\gtestresults.xml: No such file or directory
22:34:41 Could Not Find d:\jenkins\workspace\WIN-32\gtestresults.xml
```
@lrusak I know you have had enough of Windows, I'll do the fix if @Rechi or you can explain what the problem is

**Pull request merged**